src/utils/DateUtils.ts

```diff
@@ -29,138 +157,236 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
   predefinedFormat: string = undefined;
 }
 
+/**
+ * The `DateUtils` class exposes methods to convert strings, numbers and date objects to standard ISO 8601 Date objects,
+ * using the correct culture, language and format. It also offers methods to convert date objects to strings.
+ */
 export class DateUtils {
-  static convertFromJsonDateIfNeeded(date: string): Date;
-  static convertFromJsonDateIfNeeded(date: number): Date;
-  static convertFromJsonDateIfNeeded(date: Date): Date;
+  // This function is used to call convertToStandardDate for legacy reasons. convertFromJsonDateIfNeeded was refactored to
+  // convertToStandardDate, which would be a breaking change otherwise.
   static convertFromJsonDateIfNeeded(date: any): Date {
+    return DateUtils.convertToStandardDate(date);
+  }
+
+  /**
+   * Tries to parse an argument of any type to a standard Date object.
+   * @param date The value to parse. Can be of any type (string, number, Date, etc.).
+   * @returns {any} The parsed Date object, or `Invalid Date` if the `date` argument was not recognized as a valid date.
+   */
+  static convertToStandardDate(date: any): Date {
     if (_.isDate(date)) {
-      return date;
+      return moment(date).toDate();
     } else if (date !== null && !isNaN(Number(date))) {
-      return new Date(Number(date));
+      return moment(Number(date)).toDate();
     } else if (_.isString(date)) {
-      return new Date(<string>date.replace('@', ' '));
-    } else {
-      return undefined;
+      const formats = ['YYYY/MM/DD@HH:mm:ssZ', moment.ISO_8601];
+      const dateMoment = moment(date, formats);
+      return dateMoment.toDate();
```

Comment:
This will still fail silently. I think the best way to handle this would be to still return undefined, so that if this is called from a template,it still renders, but to also log a warning with the date that failed to be converted.